src/derive/stages/attributes.rs

```diff
@@ -121,14 +121,28 @@ impl Attributes {
     ) -> Vec<RawTransaction> {
         let mut transactions = Vec::new();
 
+        // L1 info (attributes deposited) transaction, present in every block
         let attributes_tx = self.derive_attributes_deposited(l1_info, input.timestamp);
         transactions.push(attributes_tx);
 
+        // User deposit transactions, present in the first block of every epoch
         if self.sequence_number == 0 {
             let mut user_deposited_txs = self.derive_user_deposited();
             transactions.append(&mut user_deposited_txs);
         }
 
+        // Optional Ecotone upgrade transactions
```

Comment:
Are these optional? I thought we had to deploy the new contracts and the code looks like it will always trigger.

Reply:
Poor naming on my end :)